src/rpc/methods/eth/filter/mod.rs

```diff
@@ -365,32 +365,20 @@ impl EthEventHandler {
         Ok(())
     }
 
-    pub async fn collect_chain_events<DB: Blockstore + Send + Sync + 'static>(
+    /// Gets events by event root.
+    pub fn get_events_by_event_root<DB: Blockstore + Send + Sync + 'static>(
         ctx: &Ctx<DB>,
-        tipset: &Tipset,
         events_root: &Cid,
     ) -> anyhow::Result<Vec<Event>> {
-        let state_events = ctx
-            .state_manager
-            .tipset_state_events(tipset, Some(events_root))
-            .await?;
-
-        ensure!(state_events.roots.len() == state_events.events.len());
-
-        let filtered_events = state_events
-            .roots
-            .into_iter()
-            .zip(state_events.events)
-            .filter(|(cid, _)| cid.as_ref() == Some(events_root))
-            .map(|(_, v)| v);
-
-        let chain_events = filtered_events
-            .into_iter()
-            .flat_map(|events| events.into_iter())
-            .map(Into::into)
-            .unique()
-            .collect();
+        let state_events =
+            match StampedEvent::get_events(ctx.chain_store().blockstore(), events_root) {
+                Ok(e) => e,
+                Err(e) => {
+                    return Err(anyhow::anyhow!("load events amt: {}", e));
```

Comment:
nit
```suggestion
                    bail!("load events amt: {e}");
```

Reply:
The error doesn't make much sense but I was trying to match lotus API level [error](https://github.com/filecoin-project/lotus/blob/7538c4740a41d993744307f756aab5966467cf55/node/impl/full/chain.go#L780) , hence this